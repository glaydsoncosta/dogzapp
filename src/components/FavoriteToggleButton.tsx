import React from 'react';
import { StyleSheet, TouchableWithoutFeedback, View } from 'react-native';
import FastImage from 'react-native-fast-image';
import { assets } from '../assets';

type FavoriteToggleButtonProps = {
  checked: boolean;
  onPress: () => void;
};

export default function FavoriteToggleButton(props: FavoriteToggleButtonProps) {
  const { checked, onPress } = props;
  return (
    <TouchableWithoutFeedback onPress={onPress}>
      <View style={styles.mainContainer}>
        <FastImage
          source={checked ? assets.icons.misc.favoriteToggleButtonChecked : assets.icons.misc.favoriteToggleButtonUnchecked}
          resizeMode={FastImage.resizeMode.contain}
          style={styles.favoriteIcon}
        />
      </View>
    </TouchableWithoutFeedback>
  );
}

const styles = StyleSheet.create({
  mainContainer: {
    justifyContent: 'center',
    alignItems: 'center'
  },
  favoriteIcon: {
    width: 24,
    height: 24
  }
});
