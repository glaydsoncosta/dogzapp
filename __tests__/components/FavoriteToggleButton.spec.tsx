import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import FavoriteToggleButton from '../../src/components/FavoriteToggleButton';

test('Should render FavoriteToggleButton component CHECKED', () => {
  const snapShot = renderer.create(<FavoriteToggleButton checked={true} onPress={() => {}} />).toJSON();
  expect(snapShot).toMatchSnapshot();
});

test('Should render FavoriteToggleButton component UNCHECKED', () => {
  const snapShot = renderer.create(<FavoriteToggleButton checked={false} onPress={() => {}} />).toJSON();
  expect(snapShot).toMatchSnapshot();
});
