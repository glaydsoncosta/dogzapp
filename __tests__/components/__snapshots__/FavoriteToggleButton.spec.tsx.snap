// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render FavoriteToggleButton component CHECKED 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 24,
          "width": 24,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="contain"
      source={
        {
          "testUri": "../../../src/assets/icons/favorites_toggle_checked.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;

exports[`Should render FavoriteToggleButton component UNCHECKED 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      [
        {
          "overflow": "hidden",
        },
        {
          "height": 24,
          "width": 24,
        },
      ]
    }
  >
    <FastImageView
      defaultSource={null}
      resizeMode="contain"
      source={
        {
          "testUri": "../../../src/assets/icons/favorites_toggle_unchecked.png",
        }
      }
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
</View>
`;
